node/cmd/guardiand/node.go

```diff
@@ -1083,6 +1087,9 @@ func runNode(cmd *cobra.Command, args []string) {
 			chainObsvReqC[vaa.ChainIDOptimism] = make(chan *gossipv1.ObservationRequest, observationRequestBufferSize)
 			optimismWatcher := evm.NewEthWatcher(*optimismRPC, optimismContractAddr, "optimism", common.ReadinessOptimismSyncing, vaa.ChainIDOptimism, lockC, nil, chainObsvReqC[vaa.ChainIDOptimism], *unsafeDevMode)
 			optimismWatcher.SetL1Finalizer(ethWatcher)
+			if err := optimismWatcher.SetRootChainParams(*optimismCtcRpc, *optimismCtcContractAddress); err != nil {
```

Comment:
SetRootChainParams() allows the two parameters to be unset because that is valid for Polygon testnet. In the case of Optimism, that is not valid, is it? Should you ensure that those parameters are set before this?

Reply:
Fixed